tradebot/utils/rate_limiter.py

```
@@ -0,0 +1,389 @@
+This module provides the RateLimiter class which handles:
+- Daily quota enforcement (800 requests/day for free tier)
+- Request spacing with minimum intervals
+- Persistent quota storage using SQLite
+- Quota reset at daily boundaries
+- Critical quota protection to prevent API limit exceeded errors
+"""
+
+import sqlite3
+import time
+import threading
+from datetime import datetime, date
```

Ruff (0.11.9)

`datetime.datetime` imported but unused

Remove unused import: `datetime.datetime`

(F401)